sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -256,6 +259,66 @@ def invalidate_environment(self, name: str) -> None:
             where=filter_expr,
         )
 
+    @transactional()
+    def delete_expired_snapshots(self) -> t.List[SnapshotTableCleanupTask]:
+        current_ts = now_timestamp(minute_floor=False)
+
+        expired_query = (
+            exp.select("name", "identifier", "version")
+            .from_(self.snapshots_table)
+            .where(exp.column("expiration_ts") <= current_ts)
+        )
+        expired_candidates = {
+            SnapshotId(name=name, identifier=identifier): SnapshotNameVersion(
+                name=name, version=version
+            )
+            for name, identifier, version in self.engine_adapter.fetchall(expired_query)
+        }
+
+        snapshots = self._get_snapshots_with_same_version(set(expired_candidates.values()))
```

Comment:
is there anyway to batch this as well for the worst case?

Reply:
What do you mean by batch here?